physical/consul.go

```diff
@@ -513,7 +513,13 @@ shutdown:
 // checkID returns the ID used for a Consul Check.  Assume at least a read
 // lock is held.
 func (c *ConsulBackend) checkID() string {
-	return "vault-sealed-check"
+	return fmt.Sprintf("vault-sealed-check:%s:%d", c.advertiseHost, c.advertisePort)
```

Comment:
@slackpad 
We are deregistering the service upon a shutdown signal. So, it should be good. The CheckID looks like this: `vault-sealed-check:127.0.0.1:8202`. I probably am missing your point here.